### PR TITLE
GT-318 Implement Index API in V2 drivers

### PR DESCRIPTION
**Pull request by jwierzbo**

### CODE REVIEW

File: v2/arangodb/analyzers.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//
+// DISCLAIMER
+//
+// Copyright 2020 ArangoDB GmbH, Cologne, Germany
```
Comment (informalict) [inline v2/arangodb/analyzers.go:4]: in all files 2020 -> 2023

**Review by informalict — APPROVED** [review]
fix boilerplate

File: v2/arangodb/collection_indexes.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+	// The index is returned, together with a boolean indicating if the index was newly created (true) or pre-existing (false).
+	// Available in ArangoDB 3.10 and later.
+	// InvertedIndexOptions is an obligatory parameter and must contain at least `Fields` field
+	EnsureInvertedIndex(ctx context.Context, options *InvertedIndexOptions) (IndexResponse, bool, error)
+
+	// DeleteIndex deletes an index from the collection.
+	DeleteIndex(ctx context.Context, name string) error
+
+	// DeleteIndexById deletes an index from the collection.
+	DeleteIndexById(ctx context.Context, id string) error
```
Comment (nikita-vanyasin) [inline v2/arangodb/collection_indexes.go:78]:
```suggestion
	DeleteIndexByID(ctx context.Context, id string) error
```

File: v2/arangodb/collection_indexes.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+	// Deduplicate is supported by array indexes of type persistent. It controls whether inserting duplicate index
+	// values from the same document into a unique array index will lead to a unique constraint error or not.
+	// The default value is true, so only a single instance of each non-unique index value will be inserted into
+	// the index per document.
+	// Trying to insert a value into the index that already exists in the index will always fail,
+	// regardless of the value of this attribute.
+	Deduplicate *bool `json:"deduplicate,omitempty"`
+
+	// Estimates  determines if the to-be-created index should maintain selectivity estimates or not.
```
Comment (nikita-vanyasin) [inline v2/arangodb/collection_indexes.go:227]:
```suggestion
	// Estimates determines if the to-be-created index should maintain selectivity estimates or not.
```

**Review by nikita-vanyasin — APPROVED** [review]

**Review by informalict — APPROVED** [review]
lgtm

**Review by informalict — APPROVED** [review]
lgtm

**Pull request merged**